tests/test_cli_sync.py

```diff
@@ -231,3 +231,15 @@ def test_sync_ask_accepted(check_call, runner):
     runner.invoke(cli, ["--ask", "--dry-run"], input="y\n")
 
     assert check_call.call_count == 2
+
+
+def test_sync_dry_run_returns_non_zero_exit_code(runner):
+    """
+    Make sure non-zero exit code is returned when --dry-run is given.
+    """
+    with open("requirements.txt", "w") as req_in:
+        req_in.write("small-fake-a==1.10.0")
+
+    out = runner.invoke(cli, ["--dry-run"])
+
+    assert out.exit_code == 2
```

Comment:
```suggestion
    assert out.exit_code == 1
```

code `1` would be more appropriate to not clash with code `2`, which is used by click in the [usage error context](https://github.com/pallets/click/blob/123dd717439d8620d8d6be5574d2c9f007952326/click/exceptions.py#L51).